File: src/main/com/lec04/Lec04Main.java
package main.com.lec04;

import main.com.etc.JavaMoney;

public class Lec04Main {

    public static void main(String[] args) {
        JavaMoney javaMoney = new JavaMoney(2000L);
        JavaMoney javaMoney1 = new JavaMoney(1000L);

        if (javaMoney.compareTo(javaMoney1) > 0) {
            System.out.println("Money가 Money1보다 금액이 큽니다.");
        }

        JavaMoney javaMoney2 = new JavaMoney(1000L);
        JavaMoney javaMoney3 = javaMoney2;
        JavaMoney javaMoney4 = new JavaMoney(1000L);

        System.out.println(javaMoney2 == javaMoney3);
        System.out.println(javaMoney2 == javaMoney4);
        System.out.println(javaMoney2.equals(javaMoney4));

        JavaMoney javaMoney5 = new JavaMoney(1000L);
        JavaMoney javaMoney6 = new JavaMoney(2000L);

        System.out.println(javaMoney5.plus(javaMoney6));
    }
}

File: src/main/com/lec04/Lec04Main.kt
package main.com.lec04

import main.com.etc.java.JavaMoney
import main.com.etc.kotlin.Money

fun main() {
    val javaMoney = JavaMoney(2000L)
    val javaMoney1 = JavaMoney(1000L)

    if (javaMoney > javaMoney1) {
        println("Money가 Money1보다 큽니다.")
    }

    val javaMoney2 = JavaMoney(1000L)
    val javaMoney3 = javaMoney2
    val javaMoney4 = JavaMoney(1000L)

    println(javaMoney2 === javaMoney3)
    println(javaMoney2 === javaMoney4)
    println(javaMoney2 == javaMoney4)

    if (fun1() || fun2()) {
        println("본문실행")
    }

    val numbers = listOf(1, 2, 3)
    println(1 in numbers)

    val money1 = Money(1000L)
    val money2 = Money(2000L)
    println(money1 + money2)
}

fun fun1(): Boolean {
    println("fun1")
    return true
}

fun fun2(): Boolean {
    println("fun2")
    return false
}